### ISSUE TITLE
2.0.0-M.2: IllegalStateException in MergedJsonContext.findType()

### ISSUE THREAD

**Issue by juergenzimmermann** [stoicflame/enunciate#107]
Inside of a build.gradle I implemented a Gradle Task which invokes the ANT task of Enunciate. However I'm getting this stacktrace:
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':enunciate'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:62)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:158)
        at org.gradle.internal.Factories$1.create(Factories.java:22)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:155)
        at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:36)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:103)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:97)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:97)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:102)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:47)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:32)
Error executing FreeMarker template
FreeMarker template error:
A type element must be provided.

----
FTL stack trace ("~" means nesting-related):
        - Failed at: ${simpleNameFor(sample_resource.respo...  [in template "jar:file:/C:/Software/enunciate/lib/enunciate-java-xml-client-2.0.0-M.2.jar!/com/webcohesion/enunciate/modules/java_xml_client/library_description.fmt" at line 61, column 1]
----

Java stack trace (for programmers):
----
freemarker.template.TemplateModelException: [... Exception message was already printed; see it above ...]
        at com.webcohesion.enunciate.util.freemarker.SimpleNameWithParamsMethod.simpleNameFor(SimpleNameWithParamsMethod.java:65)
        at com.webcohesion.enunciate.modules.java_xml_client.SimpleNameForMethod.simpleNameFor(SimpleNameForMethod.java:42)
        at com.webcohesion.enunciate.util.freemarker.SimpleNameWithParamsMethod.exec(SimpleNameWithParamsMethod.java:60)
        at freemarker.core.MethodCall._eval(MethodCall.java:62)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
        at freemarker.core.BuiltInForString._eval(BuiltInForString.java:26)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
        at freemarker.core.DollarVariable.accept(DollarVariable.java:40)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment.process(Environment.java:290)
        at freemarker.template.Template.process(Template.java:312)
        at com.webcohesion.enunciate.modules.java_xml_client.EnunciateJavaXMLClientModule.processTemplate(EnunciateJavaXMLClientModule.java:321)
        at com.webcohesion.enunciate.modules.java_xml_client.EnunciateJavaXMLClientModule.readLibraryDescription(EnunciateJavaXMLClientModule.java:544)
        at com.webcohesion.enunciate.modules.java_xml_client.EnunciateJavaXMLClientModule.generateClientSources(EnunciateJavaXMLClientModule.java:269)
        at com.webcohesion.enunciate.modules.java_xml_client.EnunciateJavaXMLClientModule.call(EnunciateJavaXMLClientModule.java:117)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
        at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
        at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
        at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
        at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
        at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
        at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
        at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
        at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
        at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:77)
        at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:47)
        at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
        at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
        at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
        at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:170)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
        at org.gradle.launcher.Main.doAction(Main.java:33)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
Caused by: : java.lang.RuntimeException: java.lang.IllegalStateException
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:116)
        at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:77)
        at org.gradle.api.internal.project.ant.BasicAntBuilder.doInvokeMethod(BasicAntBuilder.java:92)
        at build_2ut0rf5qzbi9vh7ros7x3n3b7$_run_closure19.doCall(C:\Users\ziju0001\Documents\NetBeansProjects\shop\build.gradle:647)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:554)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:535)
        at org.gradle.api.internal.tasks.TaskMutator$1.execute(TaskMutator.java:77)
        at org.gradle.api.internal.tasks.TaskMutator$1.execute(TaskMutator.java:73)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
        ... 60 more
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException
        at com.sun.tools.javac.main.Main.compile(Main.java:553)
        at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
        at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
        at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:609)
        at com.webcohesion.enunciate.EnunciateTask.execute(EnunciateTask.java:143)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        ... 69 more
Caused by: java.lang.IllegalStateException
        at com.webcohesion.enunciate.modules.java_json_client.MergedJsonContext.findType(MergedJsonContext.java:55)
        at com.webcohesion.enunciate.modules.java_json_client.SimpleNameForMethod.simpleNameFor(SimpleNameForMethod.java:31)
        at com.webcohesion.enunciate.util.freemarker.SimpleNameWithParamsMethod.exec(SimpleNameWithParamsMethod.java:60)
        at freemarker.core.MethodCall._eval(MethodCall.java:62)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
        at freemarker.core.BuiltInForString._eval(BuiltInForString.java:26)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
        at freemarker.core.DollarVariable.accept(DollarVariable.java:40)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment.process(Environment.java:290)
        at freemarker.template.Template.process(Template.java:312)
        at com.webcohesion.enunciate.modules.java_json_client.EnunciateJavaJSONClientModule.processTemplate(EnunciateJavaJSONClientModule.java:245)
        at com.webcohesion.enunciate.modules.java_json_client.EnunciateJavaJSONClientModule.readLibraryDescription(EnunciateJavaJSONClientModule.java:442)
        at com.webcohesion.enunciate.modules.java_json_client.EnunciateJavaJSONClientModule.generateClientSources(EnunciateJavaJSONClientModule.java:193)
        at com.webcohesion.enunciate.modules.java_json_client.EnunciateJavaJSONClientModule.call(EnunciateJavaJSONClientModule.java:112)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
        at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
        at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
        at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
        at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
        at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
        at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
        at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
        at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting on
Next value: com.webcohesion.enunciate.EnunciateContext.class
        at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
        at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
        ... 20 more
```

**Comment by stoicflame** [comment]
Thanks. I'll take a look for milestone 3.

**Comment by stoicflame** [comment]
Thanks again. Fixed at 5e5ed13.

**Comment by stoicflame** [comment]
[Milestone 3 is now available for you to confirm the fix](https://github.com/stoicflame/enunciate/releases/tag/v2.0.0-M.3).
